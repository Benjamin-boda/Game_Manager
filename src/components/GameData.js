import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import AddGame from "./AddGame";

const GameData = () => {
    const {title , platform} = useParams()
    const [data, setData] = useState([])
    const [displayAddGame, setDisplayAddGame] = useState(false)

    useEffect(() => {
        async function getChickenApi() {
            const fetchUrl = await fetch(`https://chicken-coop.p.rapidapi.com/games/${title}?platform=${platform}`, {
                "method": "GET",
                "headers": {
                  "x-rapidapi-host": "chicken-coop.p.rapidapi.com",
                  "x-rapidapi-key": "769c8e5236mshb7b9fd0b7a97d8fp181f53jsn56dac5b71106"
                }})
            const json = await fetchUrl.json()
            return json.result    
        }

        getChickenApi()
        .then(res => {
            setData(res)
        })
    }, [{title, platform}])

    const displayAddComponent = () => {
        setDisplayAddGame(!displayAddGame)
    }

    return (
        <div className="content-container">
            <img className="gamedata__image" src={data.image}/>
            <h2 className="gamedata__title">{data.title}</h2>       
            <p className="gamedata__description">{data.description}</p>
            <div className="gamedata__infos">
                <p className="gamedata__infos__item">Release date : {data.releaseDate}</p>
                <p className="gamedata__infos__item">Platform : {platform}</p>
                <p className="gamedata__infos__item">Developer : {data.developer}</p>
            </div>    
            <h3 className="gamedata__score">Score : {data.score}/100</h3>  
            <div className="gamedata__button">
                <button className="gamedata__infos__button" onClick={displayAddComponent}>{displayAddGame ? "Don't add" : "Add game"}</button>
            </div>
            {displayAddGame ? <AddGame data={data} platform={platform} displayAddComponent={displayAddComponent}/> : undefined}
        </div>
    )
}

export default GameData;