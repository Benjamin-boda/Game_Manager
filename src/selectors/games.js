import moment from "moment";

//Get visible games

const getVisibleGames = (games, filters) => {
    return games.filter((game) => {
        const consoleMatch = game.platform.toLowerCase().includes(filters.console.toLowerCase());
        const textMatch = game.title.toLowerCase().includes(filters.text.toLowerCase());
        const statusMatch = game.played.toLowerCase().includes(filters.status.toLowerCase());

        return textMatch && statusMatch && consoleMatch;
    })
    .sort((a, b) => {
        if (filters.sortBy === "newDate") {
            return a.boughtAt < b.boughtAt ? 1 : -1;
        } else if (filters.sortBy === "oldDate") {
            return a.boughtAt < b.boughtAt ? -1 : 1;
        };
    })
};

export default getVisibleGames;