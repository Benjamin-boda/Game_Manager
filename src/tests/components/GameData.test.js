import GameData from "../../components/GameData";
import React from "react";
import { mount } from "enzyme";
import games from "../fixtures/games";
import { Provider } from "react-redux";
import * as redux from 'react-redux'
import thunk from 'redux-thunk'
import configureMockStore from "redux-mock-store";
import { shallowToJson } from "enzyme-to-json";
import { MemoryRouter, Switch } from 'react-router-dom'


const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

test("should render AddGame page correctly", () => {
    const wrapper = shallowToJson(mount(
        <Provider store={store}>
            <MemoryRouter  initialEntries={['/']} keyLength={0}>
               <GameData data={games[0]} displayAddGame={false}/> 
            </MemoryRouter>
        </Provider>
        ))
    expect(wrapper).toMatchSnapshot();
});

// test("should open addGame", () => {
//     const displayAddComponent = jest.fn()
//     const wrapper = mount(
//         <Provider store={store}>
//             <MemoryRouter  initialEntries={['/']} keyLength={0}>
//                <GameData data={games[0]} displayAddComponent={displayAddComponent} /> 
//             </MemoryRouter>
//         </Provider>
//     )

//     wrapper.find("button").simulate("click");
//     expect(displayAddComponent).toHaveBeenCalled(); 
// });