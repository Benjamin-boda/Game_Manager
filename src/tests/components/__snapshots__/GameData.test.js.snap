// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddGame page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <GameData
        data={
          Object {
            "boughtAt": 120,
            "id": "1",
            "image": "",
            "platform": "switch",
            "played": "",
            "title": "sonic",
          }
        }
        displayAddGame={false}
      >
        <div>
          <img />
          <h2 />
          <h3 />
          <p />
          <p>
             
             
          </p>
          <button
            onClick={[Function]}
          >
            Add game
          </button>
        </div>
      </GameData>
    </Router>
  </MemoryRouter>
</Provider>
`;
